### PR TITLE
Support iCloud Library photos.

### PR DESCRIPTION
**Pull request by Wisors**
ChattoAdditions supports iCloud stored photos now. 
Whenever someone wants to send a photo from iCloud it would be downloaded with a circular progress indicator applied to a requested photo's cell.

### CODE REVIEW

File: ChattoAdditions/Source/Input/Photos/Photo/PhotosInputCellProvider.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    private var previewRequests = [Int: PhotosInputDataProviderImageRequestProtocol]()
+    private var fullImageRequests = [Int: PhotosInputDataProviderImageRequestProtocol]()
+    private func configureCell(_ cell: PhotosInputCell, at indexPath: IndexPath) {
+        if let request = self.previewRequests[cell.hash] {
+            self.previewRequests[cell.hash] = nil
+            self.dataProvider.cancelImageRequest(request)
+        }
+        self.fullImageRequests[cell.hash] = nil
+        cell.hideProgressView()
```
Comment (AntonPalich) [inline ChattoAdditions/Source/Input/Photos/Photo/PhotosInputCellProvider.swift]: Can you do it in `prepareForReuse` function? Cell state should be clean before reuse.

File: ChattoAdditions/Source/Input/Photos/PhotosInputWithPlaceholdersDataProvider.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    case error(Error?)
+
+    var image: UIImage? {
+        guard case let .success(resultImage) = self else { return nil }
+        return resultImage
+    }
+}
+
+protocol PhotosInputDataProviderImageRequestProtocol: class {
+    var isFullImageRequest: Bool { get }
```
Comment (AntonPalich) [inline ChattoAdditions/Source/Input/Photos/PhotosInputWithPlaceholdersDataProvider.swift]: Can you replace this variable with `func cancel()` that will hide implementation details about requests cancellation? This variable is only used to distinguish preview and full photos requests cancellation.

Comment (Wisors) [reply ChattoAdditions/Source/Input/Photos/PhotosInputWithPlaceholdersDataProvider.swift]: Great suggestion! Done.

File: ChattoAdditions/Source/Input/Photos/Photo/PhotosInputCellProvider.swift (hunk 1 above)
Comment (Wisors) [reply ChattoAdditions/Source/Input/Photos/Photo/PhotosInputCellProvider.swift]: 👌

**Pull request merged**